src/index.js

```diff
@@ -26,38 +37,38 @@ mh.digest = function (buf, func, length) {
   return digest
 }
 
-mh.createHash = function (func, length) {
+mh.createHash = function(func: Name | Code): Hash {
```

Comment:
This is aside comment more on the API design. I would very much encourage to either have two different functions one that works with codes the other with names or better yet settle on canonical representation and make coercion a consumer's concern. The fact that IPFS libs tend to take unions of things made by far most difficult to figure out what's being passed around (although types would solve this) and also choose representation to use myself.

Reply:
@vmx I think we should open up an issue around this. Something that could be added in another PR.

Reply:
Yes. Please open issues if you find weirdness in the API. Adding types is surely a good reason to review the APIs.